### PR TITLE
Ft oauth2 provider

### PR DESCRIPTION
**Pull request by esirK**
## Description

Adds OAuth2 provider

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### Create Application
Screenshot 2021-06-03 at 16 05 18

#### Login to ActNow app 
https://808e09c1938f.ngrok.io/o/authorize?client_id=bK0p0gKpB9W5yb1RPp6rkhQ2N8znyrNKr2UOJqvh&response_type=code

Screenshot 2021-06-03 at 16 22 11

#### Authorize application
Screenshot 2021-06-03 at 16 22 52

#### Get redirected to the app being authorized 
https://808e09c1938f.ngrok.io/promises/?code=3UTY26K3xBDjnau5PQFkQrn7siusYE

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by esirK** [comment]
Ola @kilemensi 
I've moved the CORS issues to nginx. Where should we host the app for me to be able to test with dokku?

Also, out of curiosity, other than the fact that dokku also uses Nginx for deployment, what’s the benefit of handling the `CORS` issues at Nginx level?

**Comment by kilemensi** [comment]
> Ola @kilemensi
> I've moved the CORS issues to nginx. Where should we host the app for me to be able to test with dokku?
> 
> Also, out of curiosity, other than t...

Essentially same reason as preferring to serve static assets vs nginx rather than your app be it in python / node / etc. Web servers are just better at handling non-dynamic stuff. But in our case, given the fact that we're going to have nginx in front of Django, it means CORS headers may never reach Django app if we don't explicit allow / forward them at nginx level. I'd rather we handle them there than just forward to Django and waste more processing power.

1. We need something like [this](https://enable-cors.org/server_nginx.html), but
2. For Dokku to work with this, we need to include it like we did [here](https://github.com/CodeForAfrica/HURUmap-PostGraphile-Dokku/blob/main/nginx.conf.sigil). More on nginx customisation on [Dokku](https://dokku.com/docs~v0.24.9/configuration/nginx/#customizing-the-nginx-configuration).

File: actnow/settings.py [hunk 1]
```diff
@@ -64,7 +68,7 @@
 TEMPLATES = [
     {
         "BACKEND": "django.template.backends.django.DjangoTemplates",
-        "DIRS": [],
+        "DIRS": [Path.joinpath(BASE_DIR, "templates")],
```
Comment (kilemensi) [inline actnow/settings.py:70]: What's up with this?

File: requirements-all.txt [hunk 2]
```diff
@@ -1,4 +1,6 @@
+django-cors-middleware==1.5.0
```
Comment (kilemensi) [inline requirements-all.txt]: Do we still need this (and `corsheaders` Django app) if we're handling everything at nginx level?

**Review by kilemensi — APPROVED** [review]
🚀

File: requirements-all.txt (hunk 2 above)
Comment (esirK) [reply requirements-all.txt]: nop. removing it.

File: actnow/settings.py (hunk 1 above)
Comment (esirK) [reply actnow/settings.py:70]: after defaulting back to the default admin site, we needed to place the login page at the `templates/admin/login.html` folder

Comment (kilemensi) [reply actnow/settings.py:70]: Cool... but shouldn't enabling [`APP_DIRS`](https://docs.djangoproject.com/en/3.2/ref/settings/#app-dirs) be enough?

Comment (esirK) [reply actnow/settings.py:70]: yes, it does but given this `login.html` doesn't belong to the `accounts` app but to the `admin` app, putting it in this level seemed sensible.

**Pull request merged**